packages/react-components/react-list-preview/src/components/List/List.types.ts

```diff
@@ -1,17 +1,57 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import type {
+  ComponentProps,
+  ComponentState,
+  OnSelectionChangeData,
+  Slot,
+  SelectionMode,
+  SelectionItemId,
+} from '@fluentui/react-utilities';
+import type { UseArrowNavigationGroupOptions } from '@fluentui/react-components';
+import { ListSelectionState } from '../../hooks/types';
+
+export enum ListLayout {
+  Horizontal = 'horizontal',
+  Vertical = 'vertical',
+  Grid = 'grid',
+}
+
+// Component ref interface
+export type IList = {
+  selection: ListState['selection'];
+};
 
 export type ListSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'ul', 'div' | 'ol' | 'dl'>>;
```

Comment:
Looking at the supported DOM elements for list and list items, I think we need a mechanism where users are guaranteed that they are using the correct pair of HTML elements.

I fear a situation users will use `<List as="div">` but don't actually use `<ListItem as="div">

Reply:
Thats a good callout but I'm not sure its in the scope of this PR, unless we already do have such a mechanism I could implement here.

Reply:
Ended up throwing an error in the `useListItem` in case the element combination is not valid.